### ISSUE TITLE
Outgoing http requests are not instrumented with datadog tracer - Spring Boot / RestClient / Eclipse Jetty Client 12.0.5

### ISSUE THREAD

**Issue by lokesh1729** [DataDog/dd-trace-java#7300]
Hi,

Outgoing http requests are not instrumented with datadog tracer. I am using java, spring boot, restclient with jetty 12.0.5 as http client. Here's my configuration of restclient...

We have other microservices where we were using OkHttp3 and the tracing is working fine. We pull the latest tracer in our deployment configuration and the docker files are same for every microservice.

Is jetty http client not supported?

```java
@Bean
public RestClient httpClient() {
    HttpClient httpClient = new HttpClient() {
        @Override
        public Request newRequest(URI uri) {
            Request request = super.newRequest(uri);
            request = loggingService.logRequestResponse(request);
            return middleware.handle(request);
        }
    };
    httpClient.setConnectTimeout(1000);
    httpClient.setIdleTimeout(15000);
    JettyClientHttpRequestFactory requestFactory = new JettyClientHttpRequestFactory(httpClient);
    requestFactory.setReadTimeout(Duration.ofMillis(60000));

    return RestClient.builder()
            .baseUrl(baseUrl)
            .requestFactory(requestFactory)
            .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
            .defaultHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
            .build();
}
```

**Comment by amarziali** [comment]
👋 Hi @lokesh1729 thanks for having opened the issue. Indeed jetty 12 is not yet supported. However we're (right now) working at a PR to fix it so it will be available shortly

**Comment by lokesh1729** [comment]
Hi Andrea,Thanks for the update. Any rough timelines you can share? so that we will plan our work accordingly.- LOKESH SANAPALLI- https://lokesh1729.com    ---- On Wed, 10 Jul 2024 21:48:05 +0530  ***@***.******@***.***> wrote ----  
👋 Hi @lokesh1729 thanks for having opened the issue. Indeed jetty 12 is not yet supported. However we're (right now) working at a PR to fix it so it will be available shortly

**Comment by amarziali** [comment]
The PR to support this version has just been merged. It will be available in the next release (1.38.0) foresee for start of August
